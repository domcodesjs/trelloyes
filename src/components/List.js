import React from 'react';
import Card from './Card';

const List = ({ header, cardIds, allCards }) => {
  return (
    <section className='list'>
      <header>
        <h2>{header}</h2>
      </header>
      <div className='list-cards'>
        {cardIds.map((id) => (
          <Card
            key={id}
            title={allCards[id].title}
            content={allCards[id].content}
          ></Card>
        ))}
      </div>
      <button type='button' className='list-add-card'>
        + Add Random Card
      </button>
    </section>
  );
};

export default List;
