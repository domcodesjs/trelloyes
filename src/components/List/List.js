import React from 'react';
import PropTypes from 'prop-types';
import Card from '../Card/Card';

const List = ({ header, cardIds, allCards, addCard, listId }) => {
  return (
    <section className='list'>
      <header>
        <h2>{header}</h2>
      </header>
      <div className='list-cards'>
        {cardIds.map((id) => (
          <Card
            key={id}
            title={allCards[id].title}
            content={allCards[id].content}
          ></Card>
        ))}
      </div>
      <button
        type='button'
        className='list-add-card'
        onClick={() => addCard(listId)}
      >
        + Add Random Card
      </button>
    </section>
  );
};

List.propTypes = {
  header: PropTypes.string.isRequired,
  cardIds: PropTypes.array.isRequired,
  allCards: PropTypes.object.isRequired
};

export default List;
