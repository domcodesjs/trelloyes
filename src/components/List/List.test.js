import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import store from '../../store';
import List from './List';

const cardIds = ['a', 'b', 'e', 'f', 'g', 'j', 'l', 'm'];
const allCards = store.allCards;

describe('List Component', () => {
  // Smoke Test
  it('renders without crashing', () => {
    const section = document.createElement('section');
    ReactDOM.render(
      <List header='Header' cardIds={cardIds} allCards={allCards} />,
      section
    );
    ReactDOM.unmountComponentAtNode(section);
  });

  // Snapshot Test
  it('renders the List UI as expected', () => {
    const tree = renderer
      .create(<List header='Header' cardIds={cardIds} allCards={allCards} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
