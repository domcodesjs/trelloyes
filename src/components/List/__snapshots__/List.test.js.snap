// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component renders the List UI as expected 1`] = `
<section
  className="list"
>
  <header>
    <h2>
      Header
    </h2>
  </header>
  <div
    className="list-cards"
  >
    <div
      className="card"
    >
      <div>
        <h3>
          First card
        </h3>
        <button
          type="button"
        >
          delete
        </button>
      </div>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="card"
    >
      <div>
        <h3>
          Second card
        </h3>
        <button
          type="button"
        >
          delete
        </button>
      </div>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="card"
    >
      <div>
        <h3>
          Fifth card
        </h3>
        <button
          type="button"
        >
          delete
        </button>
      </div>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="card"
    >
      <div>
        <h3>
          Sixth card
        </h3>
        <button
          type="button"
        >
          delete
        </button>
      </div>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="card"
    >
      <div>
        <h3>
          Seventh card
        </h3>
        <button
          type="button"
        >
          delete
        </button>
      </div>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="card"
    >
      <div>
        <h3>
          Tenth card
        </h3>
        <button
          type="button"
        >
          delete
        </button>
      </div>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="card"
    >
      <div>
        <h3>
          Twelfth card
        </h3>
        <button
          type="button"
        >
          delete
        </button>
      </div>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="card"
    >
      <div>
        <h3>
          Thirteenth card
        </h3>
        <button
          type="button"
        >
          delete
        </button>
      </div>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
  <button
    className="list-add-card"
    type="button"
  >
    + Add Random Card
  </button>
</section>
`;
